ts/features/walletV3/payment/screens/WalletPaymentOutcomeScreen.tsx

```diff
@@ -56,6 +60,10 @@ const WalletPaymentOutcomeScreen = () => {
 
   const handleClose = () => {
     navigation.popToTop();
+    if (paymentStartRoute) {
+      navigation.navigate(paymentStartRoute);
```

Comment:
What if I need to navigate to a sub-route?

Reply:
You're right, in this way, we cannot handle a sub-route. I changed the implementation and now the action `walletPaymentInitState` accepts an object with the `root` navigator and `screen` param. In this way, it's possible to navigate anywhere after the payment flow closes.

What do you think? 2102d662a244ebe8994df1c592a520945f0cd2fd

Reply:
I think the best way is to save all the current route info, like we are currently doing in the payment sagas:

```typescript
    const currentRouteName = NavigationService.getCurrentRouteName();
    const currentRouteKey = NavigationService.getCurrentRouteKey();
    if (currentRouteName !== undefined && currentRouteKey !== undefined) {
      yield* put(
        paymentInitializeEntrypointRoute({
          name: currentRouteName,
          key: currentRouteKey
        })
      );
    }
  ```
  
In this way we will be able to save the information on the route directly within the saga and not from the component that starts the payment flow.